test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -263,28 +280,57 @@ describe('Test getQueryTypeComplexity function', () => {
             expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(false); // ?
         });
 
-        xtest('with lists detrmined by arguments and variables', () => {
-            query = `Query {reviews(episode: EMPIRE, first: 3) { stars, commentary } }`;
-            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(4); // 1 Query + 3 reviews
-            variables = { first: 3 };
-            query = `Query queryVaribales($first: Int) {reviews(episode: EMPIRE, first: $first) { stars, commentary } }`;
-            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(4); // 1 Query + 3 reviews
+        test('with lists determined by arguments and variables', () => {
+            query = `query {reviews(episode: EMPIRE, first: 3) { stars, commentary } }`;
+            mockWeightFunction.mockReturnValueOnce(3);
+            expect(getQueryTypeComplexity(parse(query), {}, typeWeights)).toBe(4); // 1 Query + 3 reviews
+            expect(mockWeightFunction.mock.calls.length).toBe(1);
+            expect(mockWeightFunction.mock.calls[0].length).toBe(1);
+
+            variables = { first: 4 };
+            mockWeightFunction.mockReturnValueOnce(4);
+            query = `query queryVariables($first: Int) {reviews(episode: EMPIRE, first: $first) { stars, commentary } }`;
+            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5); // 1 Query + 3 reviews
+            expect(mockWeightFunction.mock.calls.length).toBe(2);
+            expect(mockWeightFunction.mock.calls[1].length).toBe(1);
         });
 
-        xtest('with nested lists', () => {
-            query = `
-            query { 
-                human(id: 1) { 
-                    name, 
-                    friends(first: 5) { 
+        xdescribe('with nested lists', () => {
+            test('and simple nesting', () => {
+                query = `
+                query { 
+                    human(id: 1) { 
                         name, 
-                        friends(first: 3){ 
-                            name 
+                        friends(first: 5) { 
+                            name, 
+                            friends(first: 3){ 
+                                name 
+                            } 
                         } 
-                    } 
-                }
-            }`;
-            expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(17); // 1 Query + 1 human/character +  (5 friends/character X 3 friends/characters)
+                    }
+                }`;
+                mockHumanFriendsFunction.mockReturnValueOnce(5).mockReturnValueOnce(3);
+                expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(17); // 1 Query + 1 human/character +  (5 friends/character X 3 friends/characters)
```

Comment:
The last time `variables` was set was on line 290 `variables = { first: 4 };`. Will this conflict with this test which uses the first argument but does not need variables.

Reply:
same for line 331

Reply:
Good catch. I'll replace `variables` with the empty object for clarity. And we can consider making this change in all tests that don't rely on variables. However, the `first` variable wouldn't actually be used by the weight function since the return value is mocked. This doesn't guarantee that it isn't used elsewhere by `getQueryTypeComplexity`. 

The variable would also need to be used in the query itself. However, I'll add a test case to the `weightFunction` test suite to ensure information isn't mixed up when a scalar if provided in the query but a variable using a limiting keyword also exists.